lib/src/repo_path.rs

```diff
@@ -1215,4 +1254,27 @@ mod tests {
             "x/something/{ => something}/1to2.txt"
         );
     }
+
+    #[test]
+    fn test_split_common_prefix() {
+        assert_eq!(
+            repo_path("foo/bar").split_common_prefix(repo_path("foo/bar/baz")),
+            Some((repo_path("foo/bar"), repo_path("")))
+        );
+
+        assert_eq!(
+            repo_path("foo/bar/baz").split_common_prefix(repo_path("foo/bar")),
+            Some((repo_path("foo/bar"), repo_path("baz")))
+        );
+
+        assert_eq!(
+            repo_path("foo/bar/bing").split_common_prefix(repo_path("foo/bar/baz")),
+            Some((repo_path("foo/bar"), repo_path("bing")))
+        );
+
+        assert_eq!(
+            repo_path("no/common/prefix").split_common_prefix(repo_path("foo/bar/baz")),
+            None
+        );
+    }
```

Comment:
nit: can you add test for `RepoPath::root()` and `same_path.split_common_prefix(same_path)`?